### PR TITLE
[[ Bug 21019 ]] Fix crash setting rect for orientation

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

File: engine/src/mblhandlers.cpp [hunk 1]
```diff
@@ -674,15 +674,20 @@ Exec_stat MCHandleSetFullScreenRectForOrientations(void *context, MCParameter *p
...
         if (p_parameters -> eval_argument(ctxt, &t_value))
         {
+            MCRectangle t_rect;
             bool t_have_rect = false;
-            ctxt.TryToConvertToLegacyRectangle(*t_value, t_have_rect, *t_rect);
+            ctxt.TryToConvertToLegacyRectangle(*t_value, t_have_rect, t_rect);
+            if (t_have_rect)
+            {
+                t_rect_ptr = &t_rect;
+            }
```
Comment (livecodeian) [inline engine/src/mblhandlers.cpp:690]: Declaring t_rect here means you risk the stack memory allocated to it being reused after leaving this scope - you're better declaring it in the same scope as t_rect_ptr so it's valid as long as t_rect_ptr is.

**Review by livecodeian — CHANGES_REQUESTED** [review]

File: engine/src/mblhandlers.cpp (hunk 1 above)
Comment (montegoulding) [reply engine/src/mblhandlers.cpp:690]: OK will push up a fix in a sec

**Review by livecodeian — APPROVED** [review]

**Comment by livecodeian** [comment]
@livecode-vulcan review ok 609597c

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeian ok 609597cb61a474d85c7daee9f517034153dfe680

**Comment by livecode-vulcan** [comment]
😎 test success 609597cb61a474d85c7daee9f517034153dfe680

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1415)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1431)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1405)
* try-community-universal-ios-iphoneos11.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.3/builds/8)
* try-community-universal-ios-iphonesimulator11.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.3/builds/8)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1409)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1435)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1415)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1429)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1414)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1424)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1465)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1405)

**Pull request merged**